### PR TITLE
Please review my code 

### PR DESCRIPTION
**Pull request by mahmmoudkinawy**
@mkinawy

### CODE REVIEW

File: Magazine/Interfaces/IGenericRepository.cs [hunk 1]
```diff
@@ -8,4 +8,5 @@ public interface IGenericRepository<T> where T : class
     void Add(T entity);
     void Update(T entity);
     void Delete(T entity);
+    Task<bool> SaveChangesAsync();
```
Comment (mkinawy) [inline Magazine/Interfaces/IGenericRepository.cs]: you are moving the unit of work responsibility to repositories, that's not a good idea as it variolates SRP
better to just call save changes internally w/o exposing this

**Review by mkinawy — COMMENTED** [review]
please remove db files (non source code files) from source control

**Comment by mkinawy** [comment]
@mahmmoudkinawy

File: Magazine/Views/Home/Index.cshtml [hunk 2]
```diff
@@ -1,8 +1,33 @@
...
-    <p>Learn about <a href="https://docs.microsoft.com/aspnet/core">building Web apps with ASP.NET Core</a>.</p>
+<div class="row">
+    @foreach (var post in Model)
+    {
+        <div class="col-sm-6">
+            <div class="card">
+                <div class="card-body">
+                    <h5 class="card-title text-center">@post.Title</h5>
+                    <p class="card-text">
+                        @if (post.Content.Length <= 30)
```
Comment (mkinawy) [inline Magazine/Views/Home/Index.cshtml]: post?

File: Magazine/Views/Articles/Index.cshtml [hunk 3]
```diff
@@ -1,32 +1,40 @@
...
+                    <th scope="col" class="col-3"></th>
                 </tr>
             </thead>
             <tbody>
                 @foreach (var post in Model)
                 {
                     <tr>
-                        <th class="col-5" scope="row">@Html.Raw( post.Content)</th>
-                        <td class="col-3">@post.CreatedDate.ToLocalTime()</td>
+                        <td class="col-2">@post.Title</td>
```
Comment (mkinawy) [inline Magazine/Views/Articles/Index.cshtml]: post?

File: Magazine/Controllers/HomeController.cs [hunk 4]
```diff
@@ -1,32 +1,21 @@
...
-using System.Diagnostics;
-
-namespace Magazine.Controllers
+﻿namespace Magazine.Controllers;
+public class HomeController : Controller
 {
-    public class HomeController : Controller
-    {
-        private readonly ILogger<HomeController> _logger;
+    private readonly IGenericRepository<Article> _postRepository;
```
Comment (mkinawy) [inline Magazine/Controllers/HomeController.cs]: post? really?

File: Magazine/Controllers/CategoriesController.cs [hunk 5]
```diff
@@ -16,33 +18,31 @@ public async Task<IActionResult> Create(Category category)
...
 
     [HttpPost]
     [ValidateAntiForgeryToken]
     public async Task<IActionResult> Update(Category category)
     {
         if (ModelState.IsValid)
         {
-            _unitOfWork.CategoryRepository.Update(category);
-            await _unitOfWork.SaveAsync();
+            _categoryRepository.Update(category);
```
Comment (mkinawy) [inline Magazine/Controllers/CategoriesController.cs]: are you sure you tested your changes? where is await?

File: Magazine/Controllers/CategoriesController.cs [hunk 6]
```diff
@@ -16,33 +18,31 @@ public async Task<IActionResult> Create(Category category)
     {
         if (ModelState.IsValid)
         {
-            _unitOfWork.CategoryRepository.Add(category);
-            await _unitOfWork.SaveAsync();
+            _categoryRepository.Add(category);
```
Comment (mkinawy) [inline Magazine/Controllers/CategoriesController.cs]: await 😠

File: Magazine/Controllers/ArticlesController.cs [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    [HttpPost, ActionName("Create")]
+    [ValidateAntiForgeryToken]
+    public IActionResult CreatePost(ArticleViewModel articleViewModel)
+    {
+        if (ModelState.IsValid)
+        {
+            articleViewModel.Article.CreatedDate = DateTime.Now;
+            _articleRepository.Add(articleViewModel.Article);
```
Comment (mkinawy) [inline Magazine/Controllers/ArticlesController.cs]: await

File: Magazine/Controllers/ArticlesController.cs [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        return View(articleViewModel);
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    public IActionResult Update(ArticleViewModel articleViewModel)
+    {
+        if (ModelState.IsValid)
+        {
+            _articleRepository.Update(articleViewModel.Article);
```
Comment (mkinawy) [inline Magazine/Controllers/ArticlesController.cs]: same

**Review by mkinawy — COMMENTED** [review]
disappointed 😞

File: Magazine/Controllers/ArticlesController.cs (hunk 8 above)
Comment (mahmmoudkinawy) [reply Magazine/Controllers/ArticlesController.cs]: I'm really sorry 😥

**Pull request merged**